src/viadot/sources/vid_club.py

```diff
@@ -0,0 +1,400 @@
+                df = pd.concat((df, df_page), axis=0)
+        else:
+            df = pd.DataFrame(response)
+
+        return df
+
+    def to_df(
+        self,
+        if_empty: str = "warn",
+    ) -> pd.DataFrame:
+        """
+        Looping get_response and iterating by date ranges defined in intervals.
+
+        Stores outputs as DataFrames in a list. At the end, daframes are concatenated
+        in one and dropped duplicates that would appear when quering.
+
+        Args:
+            if_empty (str, optional): What to do if a fetch produce no data.
+                Defaults to "warn
+
+        Returns:
+            pd.DataFrame: Dataframe of the concatanated data carried in the responses.
+        """
+        starts, ends = self.intervals(
+            from_date=self.from_date,
+            to_date=self.to_date,
+            days_interval=self.days_interval
+        )
+
+        dfs_list = []
+        if len(starts) > 0 and len(ends) > 0:
+            for start, end in zip(starts, ends):
+                logger.info(f"ingesting data for dates [{start}]-[{end}]...")
+                df = self.get_response(
+                    source=self.source,
+                    from_date=start,
+                    to_date=end,
+                    items_per_page=self.items_per_page,
+                    region=self.region,
+                )
+                dfs_list.append(df)
+                if len(dfs_list) > 1:
+                    df = pd.concat(dfs_list, axis=0, ignore_index=True)
+                else:
+                    df = pd.DataFrame(dfs_list[0])
+        else:
+            df = self.get_response(
+                source=self.source,
+                from_date=self.from_date,
+                to_date=self.to_date,
+                items_per_page=self.items_per_page,
+                region=self.region,
+            )
+        list_columns = df.columns[
+            df.map(lambda x: isinstance(x, list)).any()
+        ].tolist()
+        for i in list_columns:
+            df[i] = df[i].apply(lambda x: tuple(x) if isinstance(x, list) else x)
+        df.drop_duplicates(inplace=True)
+
+        if self.cols_to_drop is not None:
+            if isinstance(self.cols_to_drop, list):
+                try:
+                    logger.info(f"Dropping following columns: {self.cols_to_drop}...")
+                    df.drop(
+                        columns=self.cols_to_drop, inplace=True, errors="raise"
+                    )
+                except KeyError:
+                    logger.error(
+                        f"""Column(s): {self.cols_to_drop} don't exist in the DataFrame.
+                        No columns were dropped. Returning full DataFrame..."""
+                    )
+                    logger.info(f"Existing columns: {df.columns}")
+            else:
+                raise TypeError("Provide columns to drop in a List.")
+
+        if df.empty:
+            logger.error("No data for this date range")
```

Comment:
```suggestion
            self.logger.error("No data for this date range")
```

Reply:
I think Fabio changed logging in source on this:
# Configure logging
logging.basicConfig(level=logging.INFO)  
logger = logging.getLogger(__name__)

So probably this change is not needed anymore

Reply:
ok, but this is still incorrect, I added the correct solution as suggestion

Reply:
Done